primitives/credential.go

```diff
@@ -8,6 +8,7 @@ type CredentialSpec struct {
 	Key     string `json:"key"`
 	Name    string `json:"name,omitempty"`
 	Default string `json:"default,omitempty"`
+	Decoder string `json:"decoder,omitempty"`
```

Comment:
I wonder if this should be `encoding` instead? Whatever is more common, i guess.

Reply:
I've been over several options, other options seemed better every time I used one 😂 `encoding` `encoder` `decoding` `decoder` …

I'll have another look around next week in other projects what they use.

Reply:
naming things is hard!